xarray/core/parallel.py

```diff
@@ -325,13 +326,13 @@ def _wrapper(
         raise TypeError("kwargs must be a mapping (for example, a dict)")
 
     for value in kwargs.values():
-        if dask.is_dask_collection(value):
+        if is_duck_dask_array(value):
```

Comment:
this won't work because `obj` and `value` can be datasets.

Let's make a `is_dask_collection` method in `pycompat.py` that doesn't use the duck array check